src/GitHub.InlineReviews/Glyph/GlyphMarginVisualManager.cs

```diff
@@ -0,0 +1,227 @@
+                        continue;
+                    }
+
+                    data.SetSnapshot(snapshot);
+                    SnapshotSpan bufferSpan = data.VisualSpan.Value;
+                    if (!textView.TextViewLines.IntersectsBufferSpan(bufferSpan) || GetStartingLine(newOrReformattedLines, bufferSpan) != null)
+                    {
+                        visuals[data.GlyphType].Children.Remove(data.Glyph);
+                        continue;
+                    }
+
+                    dictionary[data.Glyph] = data;
+                    var startingLine = GetStartingLine(translatedLines, bufferSpan);
+                    if (startingLine != null)
+                    {
+                        data.SetTop(startingLine.TextTop - textView.ViewportTop);
+                    }
+                }
+
+                glyphs = dictionary;
+            }
+        }
+        static ITextViewLine GetStartingLine(IList<ITextViewLine> lines, Span span)
+        {
+            if (lines.Count > 0)
+            {
+                int num = 0;
+                int count = lines.Count;
+                while (num < count)
+                {
+                    int middle = (num + count) / 2;
+                    var middleLine = lines[middle];
+                    if (span.Start < middleLine.Start)
+                    {
+                        count = middle;
+                    }
+                    else
+                    {
+                        if (span.Start >= middleLine.EndIncludingLineBreak)
+                        {
+                            num = middle + 1;
+                            continue;
+                        }
+
+                        return middleLine;
+                    }
+                }
+
+                var line = lines[lines.Count - 1];
+                if (line.EndIncludingLineBreak == line.Snapshot.Length && span.Start == line.EndIncludingLineBreak)
+                {
+                    return line;
+                }
+            }
+
+            return null;
+        }
+
+        void OnFormatMappingChanged(object sender, FormatItemsEventArgs e)
+        {
+            if (e.ChangedItems.Contains(marginPropertiesName))
+            {
+                UpdateBackgroundColor();
+            }
+        }
+
+        void OnTextViewClosed(object sender, EventArgs e)
+        {
+            editorFormatMap.FormatMappingChanged -= OnFormatMappingChanged;
+        }
+
+        void UpdateBackgroundColor()
+        {
+            // set background color for children
+            var properties = editorFormatMap.GetProperties(marginPropertiesName);
+            if (properties.Contains("BackgroundColor"))
+            {
+                var backgroundColor = (Color)properties["BackgroundColor"];
+                ImageThemingUtilities.SetImageBackgroundColor(glyphMarginGrid, backgroundColor);
+            }
+
+            // set background color for margin
```

Comment:
Why is the code block commented out? I prefer not to have commented out code blocks in source especially if it's unclear _why_ they're commented out...